### PR TITLE
[SYCL] Small refactoring in scheduler

### PR DESCRIPTION
**Pull request by romanovvlad**
1. Change some functions to take std::shared_ptr by const reference
       rather than by value.
    2. Modify the helper for acquiring the graph lock to return a lock
       rather than take it as an out parameter.
    3. Move host task post processing code to a new scheduler API to
       avoid locking graph lock outside of the scheduler.

### CODE REVIEW

**Review by cperkinsintel — APPROVED** [review]
LGTM

File: sycl/source/detail/scheduler/scheduler.cpp [hunk 1]
```diff
@@ -472,6 +447,34 @@ void Scheduler::cleanupCommands(const std::vector<Command *> &Cmds) {
...
+    // update self-event status
+    Cmd->getEvent()->setComplete();
+
+    BlockingCmd->MEnqueueStatus = EnqueueResultT::SyclEnqueueReady;
+
+    for (const DepDesc &Dep : Deps)
+      Scheduler::enqueueLeavesOfReqUnlocked(Dep.MDepRequirement, ToCleanUp);
+  }
+  {
+    WriteLockT Lock = acquireWriteLock();
```
Comment (romanovvlad) [inline sycl/source/detail/scheduler/scheduler.cpp:473]: We are not ready for this lock yet.

Comment (romanovvlad) [reply sycl/source/detail/scheduler/scheduler.cpp]: Removed.

**Comment by romanovvlad** [comment]
> @romanovvlad, there are multiple failures in post-commit. Could you take a look, please?

https://github.com/intel/llvm/pull/7173

**Pull request merged**